x/ccv/provider/keeper/keeper.go

```diff
@@ -178,6 +178,49 @@ func (k Keeper) DeleteChainToChannel(ctx sdk.Context, chainID string) {
 	store.Delete(types.ChainToChannelKey(chainID))
 }
 
+// GetChainsInProposal gets chainId in gov proposal before voting finishes
+func (k Keeper) GetChainsInProposal(ctx sdk.Context, chainID string) (string, bool) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.ChainInProposalKey(chainID))
+	if bz == nil {
+		return "", false
+	}
+
+	return string(bz), true
+}
+
+// SetChainsInProposal sets consumer chainId in gov consumerAddition proposal in store
+// the consumer chainId is only added when the voting period for consumerAddition proposal
+// does not end.
+func (k Keeper) SetChainsInProposal(ctx sdk.Context, chainID string, proposalID uint64) {
```

Comment:
This needs to append to an array of existing chain_ids in proposal

Reply:
if this way, when deleting chainID, has to parse the `newKeys` to get the chainID.

Reply:
Yes. With your approach you would have to use an iterator to access the chain_id to get the entire list of chainIDs instead of just using a single key to store a slice of strings. Working with a slice of string is a bit more straightforward than working with iterators, so we skip a lot of difficult situations if we simply don't have it.

Why do you need a single chain_id? I don't understand your logic here.

Reply:
I see that it is possibly when doing a delete. In any case, lets go with your solution for now.